datasource/ethereum/src/block_stream.rs

```diff
@@ -1164,45 +816,22 @@ where
 
                 // Yielding blocks from reconciliation process
                 BlockStreamState::YieldingBlocks(mut next_blocks) => {
-                    match next_blocks.poll() {
+                    match next_blocks.pop_front() {
                         // Yield one block
-                        Ok(Async::Ready(Some(next_block))) => {
+                        Some(next_block) => {
```

Comment:
I think it's odd to push blocks without triggers to the subgraph instance, I think it's going to produce metrics observations of `0` for the triggers in block histogram and logs which say something like `0 triggers in blocks`. 

At a higher level, I like that that block stream would advance the subgraph pointer in this case even though at a lower level it adds more paths in the block streams code.

Reply:
I see that a block with zero triggers is still a sort of special case, but it already happens in the current code for non-final blocks and so far it's a special case that requires less complexity, the cases you mention with logs and metrics are easy to handle. I'm not fundamentally opposed to keeping `set_block_ptr_with_no_changes`, though I'm personally not a fan and @lutter also found it annoying, I'd be interested to know what @Jannis thinks.

Reply:
I like the idea of skipping irrelevant blocks more efficiently but I think `set_block_ptr_with_no_changes` unnecessarily triggered the logic that checks if subscriptions need to be updated? I'm all for dumping it _if_ it doesn't have a detrimental impact on indexing performance.

Reply:
Do we want to give the block stream a property of only emitting relevant blocks (with triggers) to the subgraph instance? It seems cleaner that way to me. If we don't want that or we don't want that in this PR, we should make sure to filter which blocks get logged and added to metrics on the subgraph instance side

Reply:
I just remembered another reason we had this: to get more frequent updates on indexing progress for any subgraph while scanning historic blocks. That is an important feature to preserve.

Reply:
I don't mind the 0 triggers log, during syncing most subgraphs will log that only once every 10k blocks. When synced, it seems relevant to log how many triggers are found on each new block, even if 0. If this is not helpful for the metrics, I can filter out.

I spent some time measuring performance vs master, I used Moloch because it has few events so it's seems like a good one for measuring overhead per processing cycle. Sometimes master was faster, sometimes the branch was faster, so any difference seems to be in the noise. Afaict the block stream is network bound once the blocks are cached, and this PR didn't change the rpc calls done in that scenario.

Reply:
Ahh, makes sense re: logs. Let's just filter out the zero trigger blocks from metrics then

Reply:
@olibearo Done, let me know if any other metrics need to check for this.